vm/systemSmartContracts/governance.go

```diff
@@ -700,22 +709,77 @@ func (g *governanceContract) closeProposal(args *vmcommon.ContractCallInput) vmc
 		g.addToAccumulatedFees(baseConfig.LostProposalFee)
 	}
 
-	err = g.eei.Transfer(args.CallerAddr, args.RecipientAddr, tokensToReturn, nil, 0)
+	err = g.eei.Transfer(generalProposal.IssuerAddress, args.RecipientAddr, tokensToReturn, nil, 0)
 	if err != nil {
 		g.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 
 	logEntry := &vmcommon.LogEntry{
 		Identifier: []byte(args.Function),
-		Address:    args.CallerAddr,
+		Address:    generalProposal.IssuerAddress,
 		Topics:     [][]byte{generalProposal.CommitHash, boolToSlice(generalProposal.Passed)},
 	}
 	g.eei.AddLogEntry(logEntry)
 
+	if !g.enableEpochsHandler.IsFlagEnabled(common.GovernanceFixesFlag) {
+		return vmcommon.Ok
+	}
+
+	g.processLastEndedNonce()
+	return vmcommon.Ok
+}
+
+func (g *governanceContract) clearEndedProposals(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	numAddresses := uint64(len(args.Arguments))
+	if numAddresses == 0 {
+		return vmcommon.Ok
+	}
+
+	err := g.eei.UseGas(numAddresses * g.gasCost.MetaChainSystemSCsCost.ClearProposal)
+	if err != nil {
+		g.eei.AddReturnMessage("not enough gas")
+		return vmcommon.OutOfGas
+	}
+
+	lastEndedNonce := g.getLastEndedNonce()
+	for i := uint64(0); i < numAddresses; i++ {
+		voter := args.Arguments[i]
+		if len(voter) != len(args.CallerAddr) {
+			g.eei.AddReturnMessage("invalid delegator address")
+			return vmcommon.UserError
+		}
+		err = g.clearUserVotes(voter, lastEndedNonce)
```

Comment:
from linter: `ineffectual assignment to err (ineffassign)`

Reply:
fixed